tests/common/plugins/sanity_check/__init__.py

```diff
@@ -287,10 +287,15 @@ def recover_on_sanity_check_failure(duthosts, failed_results, fanouthosts, local
                     infra_recovery_actions.append(failed_result['action'])
         for action in infra_recovery_actions:
             action()
-        for dut_name, dut_results in list(dut_failed_results.items()):
-            # Attempt to restore DUT state
-            recover(duthosts[dut_name], localhost, fanouthosts, nbrhosts, tbinfo, dut_results,
-                    recover_method)
+
+        is_modular_chassis = duthosts.nodes[0].get_facts().get("modular_chassis")
```

Comment:
Nit-pick: we can just use `duthosts[0].get_facts()....`

Reply:
Refined, thank for remind